src/Plugins/ITKImageProcessing/src/ITKImageProcessing/Filters/Algorithms/ITKImportFijiMontage.cpp

```diff
@@ -0,0 +1,323 @@
+    }
+
+    if(!m_Allocate)
+    {
+      FillCache();
+    }
+    else
+    {
+      return ReadImages();
+    }
+
+    return {};
+  }
+
+private:
+  // member variables
+  ITKImportFijiMontage* m_Filter;
+  DataStructure& m_DataStructure;
+  std::ifstream m_InStream;
+  const bool m_Allocate;
+  FijiCache& m_Cache;
+  const ITKImportFijiMontageInputValues* m_InputValues;
+
+  // -----------------------------------------------------------------------------
+  void ParseConfigFile()
+  {
+    bool dimFound = false;
+    bool dataFound = false;
+
+    std::string line;
+    while(std::getline(m_InStream, line))
+    {
+      line = StringUtilities::trimmed(line);
+
+      if(StringUtilities::starts_with(line, "dim =")) // found the dimensions
+      {
+        dimFound = true;
+        // Should check that the value = 2
+      }
+      if(StringUtilities::starts_with(line, "# Define the image coordinates"))
+      {
+        // Found the start of the data
+        dataFound = true;
+        break;
+      }
+      if(StringUtilities::starts_with(line, "#")) // comment line
+      {
+        continue;
+      }
+    }
+
+    if(!dimFound || !dataFound)
+    {
+      return;
+    }
+
+    // Example line to be parsed: slice_12.tif; ; (471.2965233276666, -0.522608066434236)
+    while(std::getline(m_InStream, line))
+    {
+      line = StringUtilities::trimmed(line);
+      if(line.empty())
+      {
+        continue;
+      }
+      std::vector<std::string> tokens = StringUtilities::split(line, ';');
+      if(tokens.size() != 3)
+      {
+        continue;
+      }
+      BoundsType bound;
+      std::stringstream path;
+      path << m_InputValues->inputFilePath.parent_path().string() << "/" << tokens[0];
+      bound.Filepath = fs::path(path.str());
+
+      std::string coords = StringUtilities::trimmed(tokens[2]);
+      coords = StringUtilities::replace(coords, "(", "");
+      coords = StringUtilities::replace(coords, ")", "");
+      tokens = StringUtilities::split(coords, ',');
+      if(tokens.size() != 2)
+      {
+        continue;
+      }
+      float32 x = std::stof(tokens[0]);
+      float32 y = std::stof(tokens[1]);
+      bound.Origin = Point3Df(x, y, 0.0f);
+      m_Cache.bounds.push_back(bound);
+    }
+  }
+
+  // -----------------------------------------------------------------------------
+  void FillCache()
+  {
+    // This next function will set the FileName (Partial), Row, Col for each "bound" object
+    ParseConfigFile();
+
+    Point3Df minCoord = {std::numeric_limits<float32>::max(), std::numeric_limits<float32>::max(), std::numeric_limits<float32>::max()};
+
+    // Get the meta information from disk for each image
+    for(auto& bound : m_Cache.bounds)
+    {
+      std::stringstream dcNameStream;
+      dcNameStream << m_InputValues->imagePrefix << bound.Filepath.filename().string().substr(0, bound.Filepath.filename().string().find(bound.Filepath.extension().string()));
```

Comment:
Do you just want the filename without the final extension here? If so `std::filesystem::path::stem()` does what you want.